godtossing/Pos/Ssc/GodTossing/Toss/Pure.hs

```diff
@@ -94,23 +91,19 @@ runPureToss
     -> m (a, GtGlobalState, [LogEvent])
 runPureToss gs (PureToss act) = do
     seed <- Rand.drgNew
-    let ((res, events), newGS) =
+    let ((res, newGS), events) =
             fst . Rand.withDRG seed $    -- run MonadRandom
-            (`runStateT` gs) $           -- run State
-            runNamedPureLog act          -- run NamedPureLogger
+            runNamedPureLog  $           -- run NamedPureLogger
+            (flip runStateT gs) act      -- run State
     pure (res, newGS, events)
 
 runPureTossWithLogger
     :: (WithLogger m, Rand.MonadRandom m)
     => GtGlobalState
     -> PureToss a
     -> m (a, GtGlobalState)
-runPureTossWithLogger gs (PureToss act) = do
-    seed <- Rand.drgNew
-    let unwrapLower a = pure $ sequenceAOf _1 $     -- (f a, b) -> f (a, b)
-                        fst $ Rand.withDRG seed $   -- run MonadRandom
-                        runStateT a gs              -- run State
-    (res, newGS) <- launchNamedPureLog unwrapLower act
+runPureTossWithLogger gs pt = do
+    (res, newGS, _) <- runPureToss gs pt
     return (res, newGS)
```

Comment:
It's not the same behaviour as it was before. `launchNamedPureLog` also uses `dispatchEvents` internally.

Reply:
Argh, I suspected this reshuffling was fishy, dang!

Thanks, will amend accordingly 😉 

> Also CI failed.

That's because I'm stuck on butchering my `pkgs/default.nix` every time as I'm on Darwin and for a long-standing issue regenerating the packages will put in a dependency on `hfsevent` instead of correctly picking up `hinotify`. Please bear with me! 😛

Reply:
Sorry :(